host/testdata/matching_simulation_fluctuating_adaptive.yaml

```diff
@@ -0,0 +1,48 @@
+enablearchival: false
+clusterno: 1
+messagingclientconfig:
+  usemock: true
+historyconfig:
+  numhistoryshards: 4
+  numhistoryhosts: 1
+matchingconfig:
+  nummatchinghosts: 4
+  simulationconfig:
+    tasklistwritepartitions: 0 # this doesn't matter. adaptive scaler will start from 1
+    tasklistreadpartitions: 0 # this doesn't matter. adaptive scaler will start from 1
+    forwardermaxoutstandingpolls: 1
+    forwardermaxoutstandingtasks: 1
+    forwardermaxratepersecond: 10
+    forwardermaxchildrenpernode: 20
+    localpollwaittime: 10ms
+    localtaskwaittime: 10ms
+    tasks:
+      - numtaskgenerators: 3
+        maxtasktogenerate: 8000
+        tasksproducespecovertime:
+        - taskspersecond: 10
+          tasksburst: 10
+          start: 0
```

Comment:
What if instead of having an explicit start and end we have a duration, and then we process these in order, calculating the effective start and end from the duration?

If we don't set the final end to be -1 then I think there's a risk of the test hanging forever due to not being able to dispatch the final tasks.

Reply:
good point. I didn't think too much of error cases here since it's not a config exposed to users. let me still fix to avoid such misconfigurations